py-polars/polars/lazyframe/frame.py

```diff
@@ -2602,11 +2634,60 @@ def filter(self, predicate: IntoExpr) -> Self:
         └─────┴─────┴─────┘
 
         """
-        if isinstance(predicate, list):
-            predicate = pl.Series(predicate)
+        # note: identify masks separately from predicates
+        all_predicates, boolean_masks = [], []
+        for p in predicates:
```

Comment:
All our `*args` functions support passing a single list as well, like `df.filter([pl.col('a') == 2, pl.col('b') == 3])`. I don't believe the current parsing logic supports this - we should add that!

Reply:
Done 👌